File: Level-2/47.java
class Solution {
    public String solution(String number, int k) {
        String answer = "";
        char max = '0';
        int idx = 0;
        
        for(int i = 0; i < number.length() - k; i++) {
            max = '0';
            for(int j = idx; j <= i + k; j++) {
                if(max < number.charAt(j)) {
                    max = number.charAt(j);
                    idx = j + 1;
                    if(max == '9') break;
                }
            }
            answer += max;
        }
        
        return answer;
    }
}

File: Level-2/47.js
function solution(number, k) {
    const stack = [];
    
    for(let num of number) {
        while(k > 0 && stack[stack.length - 1] < num) {
            stack.pop();
            k--;
        }
        stack.push(num);
    }
    
    stack.splice(stack.length - k, k);
    
    return stack.join('');
}
